### PR TITLE
Add library, create template resource ourselves

### PR DESCRIPTION
**Pull request by jdinthe**
Changes:
- Adds a dinky library to make the recipe cleaner
- Create template resource directly instead of guessing that one exists
- Filtering against IP addresses instead of just `node[:ipaddress]` (in library)
- Use `proxy_ip` key in template instead of blowing away `proxy_host`

### CODE REVIEW

File: recipes/server.rb [hunk 1]
```diff
@@ -26,27 +26,26 @@
 proxies = data_bag(node[:proxy][:databag]).collect do |name|
   data_bag_item(node[:proxy][:databag], name)
 end.select do |svc|
-  unless svc['proxy_port']
-    Chef::Log.warn("#{svc['id']} has no `proxy_port`")
+  if !verify(svc)
     false
+  else
+    Chef::Log.info "Configur proxy host IP for #{svc['id']}"
```
Comment (jdinthe) [inline recipes/server.rb]: Configur

File: libraries/proxy.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    def check_key(item, key)
+      item[key]
+    end
+
+    def check_hosts(item)
+      item['hosts'] || item.fetch(node[:short_environment], {})['hosts']
+    end
+
+    def ip_match?(item)
```
Comment (whilp) [inline libraries/proxy.rb]: I don't think `#ip_match` can be quite so terse. We want the following logic:

``` ruby
adressess = node.some_ohai_magic
# Accept an item if its proxy_host is any address (*) or not defined (nil).
addresses += ["*", nil]
addresses.include?(item['proxy_host'])
```

Comment (jdinthe) [inline libraries/proxy.rb]: Yep, that is the WIP part.

File: recipes/server.rb [hunk 3]
```diff
@@ -26,27 +26,26 @@
...
+    svc['proxy_host'] = svc['proxy_host'] ? ::Resolv.getaddress(svc['proxy_host']) : '*'
```
Comment (whilp) [inline recipes/server.rb]: I would do this resolution earlier so that we can pass the resolved IP in to `#verify`.

File: libraries/proxy.rb (hunk 2 above)
Comment (whilp) [inline libraries/proxy.rb]: Oh, womp -- saw that after. :/

Comment (jdinthe) [inline libraries/proxy.rb]: All good :) trying to jam in the ohai magic you came up with.

File: recipes/server.rb (hunk 3 above)
Comment (jdinthe) [inline recipes/server.rb]: Word.

File: libraries/proxy.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    end
+
+    def check_hosts(item)
+      item['hosts'] || item.fetch(node[:short_environment], {})['hosts']
+    end
+
+    def include_ip?(item)
+      ip_addrs = node[:network][:interfaces].collect{|name,i| i['addresses']}.collect{|i| i.keys}.flatten
+      ip_addrs.push '*'
+      ip_addrs.include?(item['proxy_ip']) ? true : false
```
Comment (whilp) [inline libraries/proxy.rb]: Does this actually need to be ternary? `#include?` returns true/false.

Comment (jdinthe) [inline libraries/proxy.rb]: Oops, yeah good point.

**Comment by whilp** [comment]
+1 after ternary fix. Much cleaner!

**Pull request merged**